### PR TITLE
Lstebner/rtl tests

### PR DESCRIPTION
**Pull request by lstebner**
### What this PR does

updates some enzyme based tests to use testing-library

- update tests for Home component
- update tests for Workshop component
- broke out helper for `getAvailableUpgrades` from workshop and added test for it too

### How this change can be validated

just run tests!

### Questions or concerns about this change

happy to refactor anything additional that stands out here

### CODE REVIEW

File: src/components/Workshop/Workshop.js [hunk 1]
```diff
@@ -201,6 +166,8 @@ const Workshop = ({ learnedRecipes, purchasedSmelter, toolLevels }) => {
 
 Workshop.propTypes = {
   learnedRecipes: object.isRequired,
+  purchasedSmelter: bool,
+  toolLevels: object.isRequired,
```
Comment (jeremyckahn) [inline src/components/Workshop/Workshop.js:170]: Thanks for adding these missing prop types!

File: src/components/Workshop/Workshop.js [hunk 2]
```diff
@@ -126,10 +95,6 @@ const Workshop = ({ learnedRecipes, purchasedSmelter, toolLevels }) => {
         </ul>
         <Divider />
         <ul className="card-list">
-          {/*
-          This really isn't a list, but it makes it easy to keep the UI
-          consistent.
-        */}
```
[the remark this answers is not in the document]
Comment (lstebner) [reply src/components/Workshop/Workshop.js:132]: 😅 i can put it back if you want but it _is_ a list as far as i can tell, it's just a single item list. and the pattern is repeated later in this file without the comment. i am one of those anti-comment people though (in favor of self documenting code of course), if it's helpful for you i don't have a problem putting it back though. your call

File: src/components/Workshop/Workshop.test.js [hunk 3]
```diff
@@ -1,20 +1,149 @@
...
+    },
+    'forge-recipe-2': {
+      id: 'forge-recipe-2',
+      name: 'forge recipe 2',
+      recipeType: 'FORGE',
+    },
+  },
+}))
+
+jest.mock('../Recipe', () => () => <div data-testid="Recipe">Recipe</div>)
```
[the remark this answers is not in the document]
Comment (lstebner) [reply src/components/Workshop/Workshop.test.js:44]: just wanted to mention i did start with no mock but these components needed some additional props provided to work and so i decided to mock them out instead to keep this spec focused. i also felt that selecting something like this was more informative than just counting the li elements for some of the expects. 

if we use this pattern a lot it's a good one to expose globally to our tests i think. like  "global.componentMock(componentName)"

File: src/components/Workshop/Workshop.test.js [hunk 4]
```diff
@@ -1,20 +1,149 @@
...
+      ).toBeInTheDocument()
+    })
+
+    test('renders a Recipe card for each learned recipe', () => {
+      expect(screen.getAllByTestId('Recipe')).toHaveLength(numLearnedRecipes)
+    })
+  })
+
+  describe('forge', () => {
+    describe('has not purchased smelter', () => {
```
[the remark this answers is not in the document]
Comment (lstebner) [reply src/components/Workshop/Workshop.test.js:96]: good call. i agree and will update this!

**Pull request closed without merging**